eng/mark-shipped.ps1

```diff
@@ -0,0 +1,51 @@
+[CmdletBinding(PositionalBinding=$false)]
+param ()
+
+Set-StrictMode -version 2.0
+$ErrorActionPreference = "Stop"
+
+function MarkShipped([string]$dir) {
+    $shippedFilePath = Join-Path $dir "PublicAPI.Shipped.txt"
+    $shipped = Get-Content $shippedFilePath
+    if ($null -eq $shipped) {
+        $shipped = @()
+    }
+
+    $unshippedFilePath = Join-Path $dir "PublicAPI.Unshipped.txt"
+    $unshipped = Get-Content $unshippedFilePath
+    $removed = @()
+    $removedPrefix = "*REMOVED*";
+    Write-Host "Processing $dir"
+
+    foreach ($item in $unshipped) {
+        if ($item.Length -gt 0) {
+            if ($item.StartsWith($removedPrefix)) {
+                $item = $item.Substring($removedPrefix.Length)
+                $removed += $item
+            }
+            elseif ($item -ne "#nullable enable") {
+                $shipped += $item
+            }
+        }
+    }
+
+    $shipped | Sort-Object | Where-Object { -not $removed.Contains($_) } | Out-File $shippedFilePath -Encoding Ascii
+    "" | Out-File $unshippedFilePath -Encoding Ascii
+}
+
+try {
+    Push-Location "$PSScriptRoot\..\src"
+
+    foreach ($file in Get-ChildItem -re -in "PublicApi.Shipped.txt") {
+        $dir = Split-Path -parent $file
+        MarkShipped $dir
+    }
+}
+catch {
+    Write-Host $_
```

Comment:
this should all be unnecessary, if error is raised it will be printed to the screen using the default error formatter that shows the error, and will terminate the task, unles you specified continue on error.  Is there a reason you added this?